framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/EntityFrameworkCore/AbpDbContext.cs

```diff
@@ -336,6 +349,31 @@ private void PublishEventsForTrackedEntity(EntityEntry entry)
         }
     }
 
+    protected virtual bool HasEntityEntryChanged(EntityEntry entry)
+    {
+        if (entry.State != EntityState.Modified && entry.State != EntityState.Unchanged)
+        {
+            return false;
+        }
+
+        var changed = entry.State == EntityState.Modified && entry.Properties.Any(x => x.IsModified && x.Metadata.ValueGenerated == ValueGenerated.Never);
```

Comment:
Shouldn't this condition also include OnAdd? 
```
var changed = entry.State == EntityState.Modified && entry.Properties.Any(x => x.IsModified && (x.Metadata.ValueGenerated == ValueGenerated.Never || x.Metadata.ValueGenerated == ValueGenerated.OnAdd));
```

Reply:
Yes, I will add it in another PR.

Reply:
https://github.com/abpframework/abp/pull/19093